tonic/models/vic/grid_params.py

```diff
@@ -728,8 +939,20 @@ def make_grid(grid_file, soil_file, snow_file, vegl_file, veg_file,
                      spatial_snow, july_tavg_supplied,
                      veglib_fcan, veglib_photo, blowing_snow,
                      vegparam_lai, vegparam_fcan, vegparam_albedo,
-                     lai_src, fcan_src, alb_src)
-        return nc_file
+                     lai_src, fcan_src, alb_src, carbon)
+        if state_file and nc_state_file:
```

Comment:
Good point - it's not a valid input argument combination (well, you could supply an input state file and not ask for an output state file, but why would you want to do that).  I probably should check for that at the beginning.